examples/fluentd/multiline-logs/Dockerfile

```diff
@@ -0,0 +1,21 @@
+# Be sure to update the versions to the ones you require
+FROM fluent/fluentd:v1.7.4-1.0
+
+# Required if you're not storing this config on S3.
+# In this case, you need to change the reference in the task definition as well.
+# ADD fluentd.conf /extra.conf
+
+# Use root account to install apk
+USER root
+
+# Below RUN installs the concat and cloudwatch-logs plugin.
+# You may want to adjust these to the plugins you require
+RUN apk add --no-cache --update --virtual .build-deps \
+        sudo build-base ruby-dev \
+ && sudo gem install fluent-plugin-concat \
+ && sudo gem install fluent-plugin-cloudwatch-logs \
+ && sudo gem sources --clear-all \
+ && apk del .build-deps \
+ && rm -rf /tmp/* /var/tmp/* /usr/lib/ruby/gems/*/cache/*.gem
+
+USER fluent
```

Comment:
@Till--H Just want to check- you've run this example and it works?

So one of the more arcane bits about how FireLens works is that the Fluentd process in log routing container has to run as root with UID 0.

This is because we use a unix socket to pipe logs from Docker into the log routing container (via the Fluentd Log Driver), and Docker runs as root, and the processes which connect to and open the socket have to have the same UID.

This in general isn't a problem that comes up because we auto-set the UID as 0 with a Docker flag when you're using FireLens. It shouldn't be a problem here either [because of how the current Fluentd images start up](https://github.com/fluent/fluentd-docker-image/blob/master/v1.7/alpine/entrypoint.sh#L28). Its only an issue if your Fluentd entrypoint calls into a script which then execs Fluentd as a different user.

You could remove this line though; since the user will be overridden to root anyway.

Reply:
We talked outside of GitHub; he's tested this with FireLens in ECS on EC2.